### PR TITLE
Updated log documentation with Builder pattern

### PR DESCRIPTION
**Pull request by anuptalwalkar**
GFM-172

PR blocked on changes reflected in #67, though I don't expect any changes in the builder pattern.

### CODE REVIEW

**Comment by anuptalwalkar** [comment]
@lupie

File: core/ulog/README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Logging package
 
-ulog provides an API wrapper around the logging library (zap Logger)
-The logger is instantiated as logger with default options and can be configured
-via `Configure()` API and provided YAML configuration.
+`package ulog` provides an API wrapper around the logging library (zap Logger).
```
Comment (Lupie) [inline core/ulog/README.md]: - Remove "Logger" and maybe add a link to [zap](https://github.com/uber-go/zap) OR make it zap.Logger (I can't tell which one is more relevant from zap's docs)
- Add some high-level benefit for users up here (why should anyone read about an API wrapper around zap?)

File: core/ulog/README.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+`ulog` uses builder pattern to instantiate the logger. Using `LogBuilder` user can set up
```
Comment (Lupie) [inline core/ulog/README.md]: - builder pattern --> the builder pattern

Looks like some people capitalize builder pattern. I don't think we need to, but let's put "the" in front of it whenever it comes up.

- Is ``LogBuilder`` familiar to readers, or do you need to explain what that is too?
- Proposed rewrite of second sentence: "With ``LogBuilder``, you can set up configuration, inject the logger, and…"
- I'm not sure if "log level prior to logger initialization" means you're logging the level OR injecting the log level before initialization

Basically, you're saying that you can set up the logger and apply it to whatever you want before logging begins, right? If so, I'd say something like this:

"``LogBuilder`` is a configuration tool for ``ulog``. With ``LogBuilder``, you can configure the logger, inject it wherever you need it, and set the log level before logging begins."

File: core/ulog/README.md [hunk 3]
```diff
@@ -11,21 +11,25 @@ import "go.uber.org/fx/core/ulog"
...
+  // build ulog.Log from logBuilder
+  log := lobBuilder.Build()
 
   // Use logger in your service
   log.Info("Message describing loggging reason", "key", "value")
 }
 ```
 
-Note that the log methods (`Info`,`Warn`, `Debug`) takes parameter as key value pairs (message, (key, value)...)
+Note that the log methods (`Info`,`Warn`, `Debug`) takes parameter as key value
```
Comment (Lupie) [inline core/ulog/README.md]: - takes --> take
- parameter --> parameters

File: core/ulog/README.md [hunk 4]
```diff
@@ -11,21 +11,25 @@ import "go.uber.org/fx/core/ulog"
...
+pairs (message, (key, value)...)
 
-ulog configuration can be defined in multiple ways, either by writing the struct yourself, or describing in the YAML
-and populating using config package.
+`ulog.Configuration` can be setup in multiple ways, either by initializing the struct,
```
Comment (Lupie) [inline core/ulog/README.md]: - setup --> set up
- multiple --> one of two

File: core/ulog/README.md [hunk 5]
```diff
@@ -11,21 +11,25 @@ import "go.uber.org/fx/core/ulog"
...
+or describing in the YAML and populating using `config` package.
```
Comment (Lupie) [inline core/ulog/README.md]: Describing what in the YAML? The configuration details?

File: core/ulog/README.md [hunk 6]
```diff
@@ -42,3 +46,25 @@ logging:
   stdout: true
   level: debug
 ```
+
+User can initialize their own `zap.Logger` implementation and inject into ulog.
```
Comment (Lupie) [inline core/ulog/README.md]: Are you guys alright with using "you" instead of "user," since these docs are for the user?

If so…
- User --> You
- their --> your
- ulog --> ``ulog``

File: core/ulog/README.md [hunk 7]
```diff
@@ -42,3 +46,25 @@ logging:
...
+To configure and inject `zap.Logger`, setup the logger prior to building
```
Comment (Lupie) [inline core/ulog/README.md]: - setup --> set up

File: core/ulog/README.md [hunk 8]
```diff
@@ -42,3 +46,25 @@ logging:
...
+the `ulog.Log` object
```
Comment (Lupie) [inline core/ulog/README.md]: Add period

File: core/ulog/doc.go [hunk 9]
```diff
@@ -20,32 +20,37 @@
 
 // Package ulog is the Logging package.
```
Comment (Lupie) [inline core/ulog/doc.go]: Where does this line come from?

- If we're going to monospace and lowercase ``package ulog``, let's do it in doc.go as well
- No reason to capitalize "logging"
- Instead: "``package ulog`` is UberFx's logging package" for better SEO and reading experience

File: core/ulog/README.md (hunk 2 above)
Comment (anuptalwalkar) [inline core/ulog/README.md]: How about - 
"With `LogBuilder` user can do pre-initialization set up by injecting configuration, custom logger and log level prior to building the usable `ulog.Log` object."

File: core/ulog/README.md (hunk 5 above)
Comment (anuptalwalkar) [inline core/ulog/README.md]: changed to "describing necessary `logging` configuration in the YAML "

File: core/ulog/doc.go (hunk 9 above)
Comment (anuptalwalkar) [inline core/ulog/doc.go]: This language is coming from gendoc, by compiling "Logging package" into Package ulog is the Logging package. We will need to change the tool to recompile this line into more friendlier language.
